src/transform/src/multiparam.ts

```diff
@@ -0,0 +1,371 @@
+          ) {
+            let body: BlockStatement;
+            if (node.body.kind != NodeKind.Block) {
+              body = Node.createBlockStatement([node.body], node.range);
+            } else {
+              body = node.body as BlockStatement;
+            }
+            node.signature.parameters.push(
+              Node.createParameter(
+                ParameterKind.Default,
+                Node.createIdentifierExpression(
+                  "__SUPPLIED_PARAMS",
+                  node.range,
+                  false,
+                ),
+                Node.createNamedType(
+                  Node.createSimpleTypeName("u64", node.range),
+                  [],
+                  false,
+                  node.range,
+                ),
+                null,
+                node.range,
+              ),
+            );
+            let first = true;
+            for (let i = 0; i < node.signature.parameters.length; i++) {
+              const param = node.signature.parameters[i];
+              if (param.parameterKind != ParameterKind.Optional) continue;
+              const stmt = Node.createIfStatement(
+                Node.createBinaryExpression(
+                  Token.Equals_Equals,
+                  Node.createParenthesizedExpression(
+                    Node.createBinaryExpression(
+                      Token.Ampersand,
+                      first
+                        ? ((first = false),
+                          Node.createIdentifierExpression(
+                            "__SUPPLIED_PARAMS",
+                            node.range,
+                          ))
+                        : Node.createParenthesizedExpression(
+                            Node.createBinaryExpression(
+                              Token.GreaterThan_GreaterThan,
+                              Node.createIdentifierExpression(
+                                "__SUPPLIED_PARAMS",
+                                node.range,
+                              ),
+                              newIntegerLiteral(i - 1, node.range),
+                              node.range,
+                            ),
+                            node.range,
+                          ),
+                      newIntegerLiteral(1, node.range),
+                      node.range,
+                    ),
+                    node.range,
+                  ),
+                  newIntegerLiteral(1, node.range),
+                  node.range,
+                ),
+                Node.createExpressionStatement(
+                  Node.createBinaryExpression(
+                    Token.Equals,
+                    Node.createIdentifierExpression(
+                      param.name.text,
+                      node.range,
+                      false,
+                    ),
+                    param.initializer,
+                    node.range,
+                  ),
+                ),
+                null,
+                node.range,
+              );
```

Comment:
Might this be easier to read and maintain if it were written as source text and parsed, rather than as an expression tree?

Reply:
... Not a requirement, but just something to consider.  If you want to keep it as-is, please add a block comment above it with the equivalent code or pseudocode.  Thanks.

Reply:
I left a comment at the top describing what it all does
Its better to do it this way so that the sources all hook up. Otherwise, we have to make a new source and errors will point to that non-existent file instead of the real file where errors are happening